iOS_Swift/PERSISTANCE.md

```diff
@@ -0,0 +1,152 @@
+
+Or it can be any kind of app state that you need to store. This can all be accomplished in several ways; each has its benefits and different complexity levels. I'll try to go through all of them from the simplest to the most complex.
+
+### NSUserDefaults
+
+The simplest way to store small amounts of data like user settings or session information is in the User Defaults. **NSUserDefaults** is a singleton class which already handles caching and persisting anything you add to it. You just need to use a string **key** and store any kind of object in it. It is basically a key-value store, that handles caching and on-disk-persistance on the fly for you.
+
+	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+     
+    [defaults setObject:firstName forKey:@"firstName"];
+    [defaults setObject:lastName forKey:@"lastname"];
+    
+    NSString *name = [defaults objectForKey: @"firstName"];
+
+### Property List Serialization
+
+**Property lists** are a way of expressing simple hierarchies of data. Basically, any **NSNumber**, **NSDate**, **NSData** and **NSString** contained in either a **NSArray** or **NSDictionary** (and any combination of them) is considered a property list. You can create Property Lists in XCode as a .plist file, if you have any static data that needs to be read into your app. Or you can create a property list in code then serialize it to disk, and vice versa.
+
+To save Plist compatible data types, you would do so like this: (plist is the dictionary we want to serialize & save to disk).
+
+	NSDictionary *plist = @{...}
+	
+	NSString *rootPath = ... // This should be the root document path for your app
+    NSString *path = [rootPath stringByAppendingPathComponent: @"DataStore.plist"];
+    
+    NSData *xmlData;
+    NSError *error;
+    
+    xmlData = [NSPropertyListSerialization dataWithPropertyList: plist
+                                                         format: NSPropertyListXMLFormat_v1_0
+                                                        options: 0
+                                                          error: &error];
+    
+    if (xmlData) {
+        [xmlData writeToFile: path atomically: YES];
+    }else{
+        DDLogError(@"ERROR : %@", error);
+    }
+	
+To open a Plist file, this is the way you would do it:
+
+	NSString *rootPath = ...
+    NSString *path = [rootPath stringByAppendingPathComponent: @"DataStore.plist"];
+    
+    NSData *plistData = [NSData dataWithContentsOfFile: path];
+    
+    NSError *error;
+    NSPropertyListFormat format;
+    id plist;
+    
+    plist = [NSPropertyListSerialization propertyListWithData: plistData
+                                                      options: NSPropertyListMutableContainers
+                                                       format: &format
+                                                        error: &error];
+    
+Don't forget to perform nil & error validation, which I excluded for brevity.
```

Comment:
@danlozano: :heart: examples. Maybe add another example with the nil & error validations for completeness?